cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -346,9 +345,13 @@ func (cm *certManagerClient) shouldUpgrade(desiredVersion string, objs, installO
 	// removes resources that are generated by the kubernetes API
 	// this is relevant if the versions are the same, because we compare
 	// the number of objects when version of objects are equal
-	objs = slices.DeleteFunc(objs, func(obj unstructured.Unstructured) bool {
-		return obj.GetKind() == "Endpoints" || obj.GetKind() == "EndpointSlice"
-	})
+	newObjs := []unstructured.Unstructured{}
+	for _, o := range objs {
+		if !(o.GetKind() == "Endpoints" || o.GetKind() == "EndpointSlice") {
+			newObjs = append(newObjs, o)
+		}
+	}
+	objs = newObjs
```

Comment:
```suggestion
        // skip resources generated by the Kubernetes API
	// this is relevant if the versions are the same, because we compare
	// the number of objects when version of objects are equal
	newObjs := []unstructured.Unstructured{}
	for _, o := range objs {
		if o.GetKind() == "Endpoints" || o.GetKind() == "EndpointSlice" {
		  contiune
		}
		newObjs = append(newObjs, o)
	}
	objs = newObjs
```

Reply:
+1

Reply:
hmm i guess the boolean logic there is a little confusing but it reads in less lines?